bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniVisitServiceImpl.java

```diff
@@ -33,4 +34,11 @@ public Visit getVisitSummary(String visitUuid) {
     public List<Encounter> getAdmitAndDischargeEncounters(Integer visitId) {
         return visitDao.getAdmitAndDischargeEncounters(visitId);
     }
+
+    @Override
+    public String getParentLocationNameForVisit(Location location) {
```

Comment:
what does this method intend to do? if the idea is to find the "visit location" then the logic is not enough. You need to traverse the location hierarchy and get the location that has "VIsit Location" tagged. If you want to find out the clinic/hospital name, then you need to go to the root of the tree and identify. Also isn't it better to return the "Location" Object itself and then the caller can get the name.

Reply:
Refactored to return Location Object.
Since the location with tag "Visit Location" is being added to the Visit, I thought getting the root of the location will give the clinic/hospital name. Please correct if this is wrong.

Reply:
Bahmni can have multiple visit locations, but ideally a single visit location in a hierarchy. 
So imagine a structure like

Org 
> Hospital 1 (Visit Location)
> Center 1  (Visit Location)

In this case, Hospital 1 / Center 1 are what a patient would expect. 

or say

Hospital 1 (Visit Location)
> Department 1 (Visit Location)
> Department 2 (Visit Location) 
 
In this case, Hospital 1 is the right one. 

So I would think the logic should be 
- Traverse the tree up, identify topmost visit location, if not the top of the tree

Reply:
refactored to get the location with tag Visit Location